app/controllers/api/v1/dependencies_controller.rb

```diff
@@ -0,0 +1,32 @@
+class Api::V1::DependenciesController < Api::BaseController
+  skip_before_action :verify_authenticity_token
+  GEM_REQUEST_LIMIT = 250
+
+  def index
+    return render text: "Too many gems!", status: 422 if gem_names.size > GEM_REQUEST_LIMIT
+
+    deps = gem_names.each_with_object([]) do |gem_name, deps_array|
```

Comment:
Sounds good. And then you can `to_json` or `to_marshal` it in the `respond_to` block.